atom/browser/native_window_views_win.cc

```diff
@@ -299,6 +300,59 @@ bool NativeWindowViews::PreHandleMSG(UINT message,
   }
 }
 
+void NativeWindowViews::HandleSizingEvent(WPARAM w_param, LPARAM l_param) {
+  double aspect_ratio = GetAspectRatio();
+  if (fabs(aspect_ratio) < 0.0001) {
+    return;
+  }
+
+  RECT* window_rect;
+  window_rect = reinterpret_cast<RECT*>(l_param);
+  int width = window_rect->right - window_rect->left;
+  int height = window_rect->bottom - window_rect->top;
+
+  double result_width = 0;
+  double result_height = 0;
+
+  gfx::Size win_fit_size(width, height);
+  const gfx::Size min_fit_size = GetMinimumSize();
+  win_fit_size.SetToMax(min_fit_size);
+
+  switch ((UINT)w_param) {
+    case WMSZ_LEFT:
+    case WMSZ_RIGHT:
+      result_width = win_fit_size.width();
+      result_height = static_cast<double>(result_width) / aspect_ratio;
+      break;
+    case WMSZ_TOP:
+    case WMSZ_BOTTOM:
+      result_height = win_fit_size.height();
+      result_width = static_cast<double>(result_height) * aspect_ratio;
+      break;
+    case WMSZ_TOPLEFT:
+    case WMSZ_TOPRIGHT:
+    case WMSZ_BOTTOMLEFT:
+    case WMSZ_BOTTOMRIGHT: {
+      result_width = static_cast<double>(height) * aspect_ratio;
+      result_height = height;
+
+      double temp_width = width;
+      double temp_height = static_cast<double>(width) / aspect_ratio;
+
+      if (abs(temp_width * temp_height) > abs(result_width * result_height)) {
+        result_width = temp_width;
+        result_height = temp_height;
+      }
+      break;
+    }
+    default:
```

Comment:
will this ever get reached? If it does, the result is a 0x0 window, which might be confusing :smiley: